design/covariance_plot.Rmd

```diff
@@ -0,0 +1,145 @@
+
+```{r}
+#' @description This function vectorizes the upper-diagonal elements of a symmetric matrix (e.g. the covariance matrix)
+#' and obtains the lag and time distance between pairs of observations if the time values are part of
+#' the names or are part of the matrix column/row names.
+#' @param vcov_matrix (`matrix`)\cr name of the input symmetric matrix.
+#' @return This function returns a data frame with the upper-diagonal elements or a covariance or correlation
+#' matrix. In the context of repeated measures, this matrix contains the association between pairs
+#' of measurements taken at different time points.
+#' @examples
+#' h_vectorization(vcov_matrix = cholest.cov)
+h_vectorization <- function(vcov_matrix) {
+  assert_matrix(vcov_matrix)
+  assert_true(isSymmetric(vcov_matrix))
+  diag <- upper.tri(vcov_matrix, diag = T)
+  vect <- vcov_matrix[diag]
+  timepoints <- h_get_timepoint_vars(vcov_matrix)
+  time_point_distribution <- timepoints$numeric_vcov_matrix_col_label - timepoints$numeric_vcov_matrix_row_label
+  rank_row <- as.numeric(as.factor(timepoints$numeric_vcov_matrix_row_label))
+  rank_col <- as.numeric(as.factor(timepoints$numeric_vcov_matrix_col_label))
+  lag <- rank_col - rank_row
+  if (any(is.na(lag)) | any(is.na(time_point_distribution))) warning("Verify you have used the string argument correctly or that the columns and rows of your input matrix are named as expected")
+  vect <- data.frame(Vect = vect, time_point_distribution, lag, rank_row, rank_col)
+  return(vect)
+}
+```
+
+Now we use this helper function as input for the function to create the covariance plot. 
+
+```{r}
+#' @description Plot of covariance or correlation structures as a function of lag or time. The covariance structure
+#' needs to be vectorized and lag or time distances computed
+#' @param  vcov_dataframe (`data frame`}\cr name of the data frame created by h_vectorization.
+#' @param {x_var} (`string`)\cr can be "lag" or "time_point_distribution" for lag and time distance respectively. The default value is "lag".
+#' @param legend_position (`string`)\cr denoting where the legend should be shown. Option are c("topright","topleft","bottomright","bottomleft"). Default #' value #' is "topright"
+#' @examples
+#' g_covariance(vcov_dataframe = h_vectorization(vcov_matrix = cholest.cov), x_var = "time_point_distribution")
+g_covariance <- function(vcov_dataframe, x_var = c("lag", "time_point_distribution"), ylab = "",
```

Comment:
looks really good now Julia! nice work. maybe for production later to consider: should `g_covariance` take the `vcov_matrix` or `vcov_dataframe` as input? I would only go for the latter if the user wanted to change the data frame somehow manually in between before passing it to `g_covariance`. otherwise it is more convenient to pass the covariance matrix directly and then inside call `h_vectorization`